### DISCUSSION TITLE
issue: useController shouldUnregister not working as expected

### DISCUSSION THREAD

**Discussion by neviaumi** [react-hook-form/react-hook-form#7466] [Q&A]
### Version Number

7.22.5

### Codesandbox/Expo snack

https://stackblitz.com/edit/react-vkvev9?file=src/App.js

### Steps to reproduce

1. Go to 'Add Member'
2. type 100 on 'Ownership'
3. Unselect  'SR'
4. Select 'AC'
4. Click Submit

### Expected behaviour

form value don't contain percentageOfOwnership

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
![image](https://user-images.githubusercontent.com/10513364/148536974-d46374cf-17bc-47c1-90a7-343423ef08d1.png)

> unmounted input will need to notify at either useForm, or useWatch's useEffect for hook form to verify input is unmounted from the DOM.

if the state update is not within useForm or useWatch's useEffect, you will have to invoke `unregister` manually.

**Comment by neviaumi** [comment]
@bluebill1049 

Here is my secound attempt
https://stackblitz.com/edit/react-8jmrea?file=src/App.js

Did you mean i should include return value from `useWatch` in condition?

Didn't working either.

But manually unregister is working perfectly for me.

just
```js
 const shouldAskPercentageOfOwnership = useMemo(
    () =>
      roles?.find((role) =>
        [MEMBER_ROLES.SHAREHOLDER, MEMBER_ROLES.DIRECTOR].includes(role),
      ),
    [roles],
  );
  useEffect(() => {
    if (!shouldAskPercentageOfOwnership) unregister(fieldName(index));
  }, [shouldAskPercentageOfOwnership, index, unregister]);
  if (!shouldAskPercentageOfOwnership) return null;
```

**Reply by bluebill1049** [reply]
is your component gets unmounted? if not then you should use `unregister` method.

**Reply by neviaumi** [reply]
I think `PercentageOfOwnershipFieldWrapper` is already did when roles is not `SR` and `DR` then unmount `PercentageOfOwnershipField` ? 

Or i need something special to archive that ? i feel like `unregister` is more like dirty fix
